usaspending_api/awards/v2/filters/location_filter_geocode.py

```diff
@@ -45,52 +40,58 @@ def geocode_filter_locations(scope, values, model, use_matview=False, app_name='
-            if nested_locations.get(v['country']) is None:
-                nested_locations[v['country']] = {}
-
-            # Second level of filtering is zip and state
-            # Requests must have a country+zip or country+state combination
-            if 'zip' in v:
-                nested_locations[v['country']]['zip'] = v['zip']
-            elif 'state' in v and nested_locations[v['country']].get(v['state']) is None:
-                nested_locations[v['country']][v['state']] = {'county': [], 'district': []}
-
-            # Under state a user can filter on county and/or district (congressional_code)
-            # Error will be thrown if no state value
-            if v.get('county'):
-                nested_locations[v['country']][v['state']]['county'].extend(get_fields_list('county', v['county']))
-
-            if v.get('district'):
-                nested_locations[v['country']][v['state']]['district'].extend(
-                    get_fields_list('district', v['district']))
-
-        except KeyError:
-            # This result if there is value for country or state where necessary
+        if ("country" not in v) or (("district" in v or "county" in v) and "state" not in v):
             location_error_handling(v.keys())
 
+
+def create_nested_object(values):
+    # Makes sure keys provided are valid
+    validate_location_keys(values)
+
+    nested_locations = {}
+    for v in values:
+        # First level in location filtering in country
+        # All location requests must have a country otherwise there will be a key error
+        if nested_locations.get(v['country']) is None:
+            nested_locations[v['country']] = {}
+
+        # Initialize the list
+        if 'zip' in v and not nested_locations[v['country']].get('zip'):
+            nested_locations[v['country']]['zip'] = []
+
+        # Second level of filtering is zip and state
+        # Requests must have a country+zip or country+state combination
+        if 'zip' in v:
+            # Appending zips so we don't overwrite
+            nested_locations[v['country']]['zip'].append(v['zip'])
+
+        # If we have a state, add it to the list
+        if 'state' in v and nested_locations[v['country']].get(v['state']) is None:
+            nested_locations[v['country']][v['state']] = {'county': [], 'district': []}
+
+        # Based on previous checks, there will always be a state if either of these exist
+        if v.get('county'):
+            nested_locations[v['country']][v['state']]['county'].extend(get_fields_list('county', v['county']))
+
+        if v.get('district'):
+            nested_locations[v['country']][v['state']]['district'].extend(get_fields_list('district', v['district']))
     return nested_locations
 
 
 def location_error_handling(fields):
-    # Request must have country, and can only have 3 fields,
-    # and must have state if there is county or district
+    """ Raise the relevant error for location keys. """
```

Comment:
This function can be deleted by moving the InvalidParameterException logic into `validate_location_keys()`